libsolidity/Compiler.cpp

```diff
@@ -486,9 +494,10 @@ bool Compiler::visit(ExpressionStatement const& _expressionStatement)
 	return false;
 }
 
-bool Compiler::visit(PlaceholderStatement const&)
+bool Compiler::visit(PlaceholderStatement const& _placeholderStatement)
 {
 	StackHeightChecker checker(m_context);
+	CompilerContext::LocationSetter locationSetter(m_context, &_placeholderStatement);
 	++m_modifierDepth;
 	appendModifierOrFunctionCode();
```

Comment:
I think inside `appendModifierOrFunctionCode` we should also set a new location. There might be quite some other places where we should do it.